### PR TITLE
Now can go back 3 year in usage stats

### PR DESCRIPTION
**Pull request by jfkonecn**
closes https://github.com/oncokb/oncokb-pipeline/issues/418

Here are the endpoints I identified as using the usage data.

- <http://localhost:9000/admin/usage-analysis#type=USER>
- <http://localhost:9000/admin/usage-analysis/users/8>
- <http://localhost:9000/admin/usage-analysis/resources/!api!v1!annotate!mutations!byProteinChange>
- <http://localhost:9000/companies/3>
- <http://localhost:9000/users/XXXXXXX@mskcc.org>

@zhx828 the performance bottleneck seems to be fetching the s3 files

### CODE REVIEW

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/CronJobController.java [hunk 1]
```diff
@@ -187,19 +187,6 @@ public void moveTokenStatsToS3() throws IOException {
...
-        // Deal with month summary
-        if (!usageSummary.getMonth().containsKey(time)) {
-            usageSummary.getMonth().put(time, new JSONObject());
-        }
-        if (!usageSummary.getMonth().get(time).containsKey(key)) {
-            usageSummary.getMonth().get(time).put(key, new Integer(0));
-        }
-        usageSummary.getMonth().get(time).put(key, (Integer) usageSummary.getMonth().get(time).get(key) + (Integer) count);
-    }
-
```
Comment (jfkonecn) [inline src/main/java/org/mskcc/cbio/oncokb/web/rest/CronJobController.java:202]: My changes caused a type error and this code isn't used so I chose to delete it.

File: src/main/webapp/app/pages/usageAnalysisPage/usage-analysis-utils.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+function createResourceTableRecord(
+  time: string,
+  usageSummary: Record<string, number>,
+  keyType: 'resource' | 'email',
+  isAllUsage: boolean
+): UsageRecord[] {
+  return Object.entries(usageSummary)
+    .filter(([key]) => isAllUsage || !key.startsWith('/api/private/'))
```
Comment (zhx828) [inline src/main/webapp/app/pages/usageAnalysisPage/usage-analysis-utils.ts:58]: I noticed `key.startsWith('/api/private/'))` used a lot. I know it looks like a simple logic, but can we wrap it up as a method?

**Review by zhx828 — CHANGES_REQUESTED** [review]
- For pages with mutliple tables and other info, let's use default page size 5 -> company page, user details page.
- should we add some tests for usage-analysis-utils?

**Review by zhx828 — APPROVED** [review]
Thanks！

**Pull request merged**